test/DynamoCoreWpfTests/UnitsUITests.cs

```diff
@@ -199,7 +199,7 @@ public void ForgeUnitDropdownsLoadWithMalformedData()
             //we've changed both index and name, so a new item is selected.
             Assert.AreNotEqual(DynamoUnits.Quantity.ByTypeID("autodesk.unit.quantity:force-1.0.2"), node3.CachedValue.Data);
             Assert.AreEqual(DynamoUnits.Symbol.ByTypeID("autodesk.unit.symbol:mm-1.0.1"), node4.CachedValue.Data);
-            Assert.AreEqual(DynamoUnits.Unit.ByTypeID("autodesk.unit.unit:millimeters-1.0.1"), node5.CachedValue.Data);
+            Assert.AreEqual(DynamoUnits.Unit.ByTypeID("autodesk.unit.unit:microarcseconds-1.0.1"), node5.CachedValue.Data);
```

Comment:
?

Reply:
@QilongTang @saintentropy do I have that right? I kind of remember this is a general problem with dropdowns on the Revit side, though I thought we had semi fixed it with string matching as well as index matching at some point?
I am remembering incorrectly?

Reply:
Right, to me it looks like opening a legacy graph the node is returning a different value, if the old value is still there from the list, then this could be a bug worth fixing. I would suggest checking the serialized index and node value and compare with the new graph opening

Reply:
The node in question
```      "ConcreteType": "UnitsUI.Units, UnitsUI",
      "SelectedIndex": 221,
      "SelectedString": "Millimeters AAA",
      "NodeType": "ExtensionNode",
      "Id": "2cc49d25d3b447b281a81efd6173c6cf",
      ```
It has an intentional malformed selection "Millimeters AAA" - that does not exist.
So in turn it defaults to the selected index (221) which now is different, because of the changes that went in

Reply:
I don't want to hold this up further, but I am trying to understand the following:
1. Is this failure an existing issue with all dropdowns or just these unit nodes?
2. What was this test intending to test? Does it still test that?

Reply:
1. I think this scenario could affect all dropdowns
2.  It looks to me like this test wants to cover the stability of unit drop downs. When the serialized selection exists and when it does not exist anymore. I think there is still value in it 
If a unit value is renamed, the selectedIndex could still point to the new (renamed) unit. 
This test is prone to failures if new units are added or removed.